platform-tools-sdk/cargo-build-sbf/src/main.rs

```diff
@@ -605,6 +605,21 @@ fn build_solana_package(
             .iter()
             .filter_map(|target| {
                 if target.crate_types.contains(&"cdylib".to_string()) {
+                    let other_crate = if target.crate_types.contains(&"rlib".to_string()) {
+                        Some("rlib")
+                    } else if target.crate_types.contains(&"lib".to_string()) {
+                        Some("lib")
+                    } else {
+                        None
+                    };
+
+                    if let Some(other_crate) = other_crate {
+                        warn!("Package '{}' has two crate types defined: cdylib and {}. \
+                        This setting precludes compiler optimizations. Use cdylib for programs \
```

Comment:
Can we call out the optimizations? ie "This setting precludes compiler optimizations: LTO, <OTHER_THING>, and <LAST_THING>."

Reply:
Fixed in https://github.com/anza-xyz/agave/pull/5729/commits/3da031b6c7fc07764746e2974e2001f1f288bcf3